azurelinuxagent/common/utils/shellutil.py

```diff
@@ -345,10 +345,23 @@ def quote(word_list):
 #
 _running_commands = []
 _running_commands_lock = threading.RLock()
+PARENT_PROCESS_NAME = "PARENT_PROCESS_NAME"
+AZURE_GUEST_AGENT = "AZURE_GUEST_AGENT"
 
 
 def _popen(*args, **kwargs):
     with _running_commands_lock:
+        # Add the environment variables
+        env = {}
+        if 'env' in kwargs:
+            env.update(kwargs['env'])
+        else:
+            env.update(os.environ)
+
+        # Set the marker before process start
+        env[PARENT_PROCESS_NAME] = AZURE_GUEST_AGENT
+        kwargs['env'] = env
```

Comment:
If caller sends `env` arg, use the same flags otherwise extend `os.enviorn` along with new marker flag